src/Translator.php

```diff
@@ -275,6 +279,51 @@ private function mergeYaml(): void
         $this->log('Finished merging ' . count($this->originalYaml) . ' yaml files');
     }
 
+    private $blankEnStrings;
+
+    private function removeBlankStrings(array &$sourceData)
+    {
+        $this->blankEnStrings = [];
+        foreach ($sourceData as $lang => &$data) {
+            if ($lang !== 'en') {
+                continue;
+            }
+            $this->recursiveRecordBlankStrings('', $data);
+        }
+        foreach ($sourceData as $lang => &$data) {
+            if ($lang === 'en') {
+                continue;
+            }
+            if (is_array($data)) {
+                $this->recursiveRemoveBlankStrings('', $data);
+            }
+        }
+    }
+
+    private function recursiveRecordBlankStrings(string $parentKey, array &$data)
+    {
+        foreach ($data as $key => &$val) {
+            if (is_array($val)) {
+                $this->recursiveRecordBlankStrings($parentKey . '-' . $key, $val);
+            }
+            if (empty($val)) {
+                $this->blankEnStrings[$parentKey . '-' . $key] = true;
+            }
+        }
+    }
+
+    private function recursiveRemoveBlankStrings(string $parentKey, array &$data)
+    {
+        foreach ($data as $key => &$val) {
+            if (is_array($val)) {
+                $this->recursiveRemoveBlankStrings($parentKey . '-' . $key, $val);
+            }
+            if (empty($val) && !array_key_exists($parentKey . '-' . $key, $this->blankEnStrings)) {
```

Comment:
```suggestion
            if ($val === '' && !array_key_exists($parentKey . '-' . $key, $this->blankEnStrings)) {
```
Definitely an edge case, but protecting against a case where the translation string is `'0'` which PHP considers empty.

Reply:
Updated.